src/elements/earn/pools/addLiquidity/single/AddLiquiditySingleSpaceAvailable.tsx

```diff
@@ -0,0 +1,104 @@
+import { Pool, Token } from 'services/observables/tokens';
+import {
+  checkPriceDeviationTooHigh,
+  fetchBntNeededToOpenSpace,
+  getSpaceAvailable,
+} from 'services/web3/contracts/liquidityProtection/wrapper';
+import { useCallback, useEffect, useState } from 'react';
+import { prettifyNumber } from 'utils/helperFunctions';
+import { useInterval } from 'hooks/useInterval';
+import BigNumber from 'bignumber.js';
+
+interface Props {
+  pool: Pool;
+  token: Token;
+  selectedToken: Token;
+}
+
+export const AddLiquiditySingleSpaceAvailable = ({
+  pool,
+  token,
+  selectedToken,
+}: Props) => {
+  const [showPriceDeviationWarning, setShowPriceDeviationWarning] =
+    useState(false);
+  const [spaceAvailableBnt, setSpaceAvailableBnt] = useState('');
+  const [spaceAvailableTkn, setSpaceAvailableTkn] = useState('');
+  const [bntNeeded, setBntNeeded] = useState('');
+
+  useInterval(
```

Comment:
minor: the entire fetchData definition and its usage in useInterval and useEffect can be extracted to a hook